### PR TITLE
2092 df ex

### LINKED ISSUE #2092: Fix how `Dataframe` examples look when a string is passed in
### Describe the bug

With the latest version of Gradio, you can pass in string filepaths to CSVs as examples to the `Dataframe` component. However, the examples are not rendered properly in the Examples component, as you can see here:

image

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```py
import gradio as gr
import pandas as pd
import numpy as np

examples = [["test.csv"]]

def get_max(df: pd.DataFrame) -> float:
    return df.max().max()


gr.Interface(get_max, gr.Data
Frame(type="pandas", headers=["a", "b", "c"]), gr.Number(),
             examples=examples).launch()
```

Also have the attached `test.csv` file in your directory.

[test.csv](https://github.com/gradio-app/gradio/files/9434365/test.csv)

### Screenshot

_No response_

### Logs

```shell
N/A
```

### System Info

```shell
latest
```

### Severity

annoying

### PR DESCRIPTION
**Pull request by pngwn**
Closes #2092.

(rows x cols )
- 3 x 3, no truncation
- 4+ x 3, rows truncated to 3 and fade on bottom
- 3 x 4+, cols truncated to 3 and ellipsis on right
- 4+ x 4+, both truncated to 3, fade on bottom, ellipsis on right

Looks like this now (one of each variant), the far right preview is a CSV, these are now loaded parsed and displayed with everything else. The file name is not particularly useful information (although I could add the name as well if needed):
Screenshot 2022-08-30 at 18 02 16

The final
Can be tested with the following snippet + the `test.csv` in the same folder [here](https://github.com/gradio-app/gradio/files/9434365/test.csv):

```py
import gradio as gr
import pandas as pd

examples = [
    [
        [
            [ 1, 2, 3 ],
            [ 1, 2, 3 ],
            [ 1, 2, 3 ],
        ],
    ],
    [
        [
            [ "a", "b", "c" ],
            [ "a", "b", "c" ],
            [ "a", "b", "c" ],
            [ "a", "b", "c" ],
        ]
    ],
    [
        [
            ["a", "b", "c", "d"],
            ["a", "b", "c", "d"],
            ["a", "b", "c", "d"],
        ]
    ],
    [
        [
            ["hi", "none", "1", "1"],
            ["hello", "nada", "1", "1"],
            ["hey there", "zilch", "1", "1"],
            ["heeeeeeeeey", "nothing", "1", "1"],
        ]
    ],
    ["test.csv"],
]


def get_max(df: pd.DataFrame) -> float:
    return df.max().max()


demo = gr.Interface(
    get_max,
    gr.DataFrame(type="pandas", headers=["a", "b", "c"]),
    gr.Number(),
    examples=examples,
)

if __name__ == "__main__":
    demo.launch()
```

### CODE REVIEW

File: ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte [hunk 1]
```diff
@@ -1,23 +1,69 @@
...
+	let hovered = false;
+	let loaded = Array.isArray(value);
+
+	$: if (!loaded && typeof value === "string" && /\.[a-zA-Z]+$/.test(value)) {
+		fetch(samples_dir + value)
+			.then((v) => v.text())
+			.then((v) => {
+				try {
+					if ((value as string).endsWith("csv")) {
+						value = csvParseRows(v);
```
Comment (freddyaboulton) [inline ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte:15]: Is this reading in the whole file? If so, I don't think we need to - just at most the first 5 rows and columns rights?

**Review by freddyaboulton — COMMENTED** [review]
@pngwn This is really cool! I think the new display looks great and I verified it works as expected when a dataframe is nested along with other component examples:

image

The one nit-pick I have is that the fade-out is maybe too dark when running in light mode, e.g. spaces:

image

File: ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte (hunk 1 above)
Comment (pngwn) [reply ui/packages/app/src/components/Dataset/ExampleComponents/Dataframe.svelte:15]: Yeah, that’s true. I guess some CSVs could be pretty huge.

**Comment by pngwn** [comment]
@freddyaboulton Good catch. That should be shire in light mode. Will fix.

**Comment by pngwn** [comment]
@freddyaboulton Both should be addressed now.

**Review by freddyaboulton — APPROVED** [review]
Thank you for the fixes and PR @pngwn !

**Pull request merged**